leda/utils/useMessages.ts

```diff
@@ -1,21 +1,19 @@
 import * as React from 'react';
-import type { globalDefaultTheme } from '../components/LedaProvider';
 import { LedaContext } from '../components/LedaProvider';
-import type { RecursivePartial } from '../commonTypes';
 import { globalDefaultMessages } from '../components/LedaProvider/globalDefaultMessages';
 
-export type GlobalDefaultTheme = typeof globalDefaultTheme;
-
-export type PartialGlobalDefaultTheme = RecursivePartial<GlobalDefaultTheme>;
-
-export const useMessages = <T extends keyof typeof globalDefaultMessages>(
+export const useMessages = <T extends keyof typeof globalDefaultMessages>({
+  fieldName,
+  messages: componentMessages,
+}: {
   fieldName: T,
-): typeof globalDefaultMessages[T] => {
+  messages?: typeof globalDefaultMessages[T],
+}): typeof globalDefaultMessages[T] => {
   const {
     messages: globalMessages,
   } = React.useContext(LedaContext);
 
   return React.useMemo(() => ({
-    ...globalDefaultMessages[fieldName], ...globalMessages[fieldName],
-  }), [fieldName, globalMessages]) as unknown as typeof globalDefaultMessages[T];
+    ...globalDefaultMessages[fieldName], ...globalMessages[fieldName], ...componentMessages,
+  }), [fieldName, globalMessages, componentMessages]) as unknown as typeof globalDefaultMessages[T];
```

Comment:
It might break memorization completely
You have you use a ref hack as a workaround or something similar

Reply:
How come it breaks memorization?

Reply:
It depends on the messages from props now

If I'll provide it like `<Autocomplete messages={{ title: "text" }} />` then memorization will be broken

Reply:
I think this is going to be the best solution here:

```
export const useMessages = <T extends keyof typeof globalDefaultMessages>({
  fieldName,
  messages: componentMessages,
}: {
  fieldName: T,
  messages?: typeof globalDefaultMessages[T],
}): typeof globalDefaultMessages[T] => {
  const {
    messages: globalMessages,
  } = React.useContext(LedaContext);

  return {
    ...globalDefaultMessages[fieldName], ...globalMessages[fieldName], ...componentMessages,
  };
};
```
using object literals in components' props is considered to be an anti-pattern, so it is not library's responsibility
moreover useMessages does not perform any costly calculations, so useMemo is just unnecessary optimisation here leading to unnecessary complexity when the code grows